### PR TITLE
[SPARK-48386][TESTS] Replace JVM assert with JUnit Assert in tests

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to replace `JVM assert` with `JUnit Assert` in tests.

### Why are the changes needed?
assert() statements do not produce as useful errors when they fail, and, if they were somehow disabled, would fail to test anything.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
- Manually test.
- Pass GA.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Comment by panbingkun** [comment]
This PR reference:
https://github.com/apache/spark/commit/1febd373ea806326d269a60048ee52543a76c918
https://github.com/apache/spark/pull/37772

**Comment by panbingkun** [comment]
cc @LuciferYang @dongjoon-hyun @HyukjinKwon

**Comment by LuciferYang** [comment]
I think it is necessary to create a jira

**Comment by panbingkun** [comment]
> I think it is necessary to create a jira

Done.

**Review by LuciferYang — APPROVED** [review]
+1, LGTM (pending tests)

File: common/utils/src/test/java/org/apache/spark/util/SparkLoggerSuiteBase.java [hunk 1]
```diff
@@ -127,8 +128,8 @@ public void testBasicLoggerWithException() {
         Pair.of(Level.DEBUG, debugFn),
         Pair.of(Level.TRACE, traceFn)).forEach(pair -> {
       try {
-        assert (captureLogOutput(pair.getRight()).matches(
-            expectedPatternForBasicMsgWithException(pair.getLeft())));
+        Assertions.assertTrue(captureLogOutput(pair.getRight()).matches(
```
Comment (dongjoon-hyun) [inline common/utils/src/test/java/org/apache/spark/util/SparkLoggerSuiteBase.java:131]: Why don't we import this?

Comment (panbingkun) [reply common/utils/src/test/java/org/apache/spark/util/SparkLoggerSuiteBase.java]: Updated.

File: sql/core/src/test/java/test/org/apache/spark/sql/connector/JavaAdvancedDataSourceV2WithV2Filter.java [hunk 2]
```diff
@@ -78,9 +80,9 @@ public Predicate[] pushPredicates(Predicate[] predicates) {
 
       Predicate[] unsupported = Arrays.stream(predicates).filter(f -> {
         if (f.name().equals(">")) {
-          assert(f.children()[0] instanceof FieldReference);
+          Assertions.assertInstanceOf(FieldReference.class, f.children()[0]);
```
[the remark this answers is not in the document]
Comment (panbingkun) [reply sql/core/src/test/java/test/org/apache/spark/sql/connector/JavaAdvancedDataSourceV2WithV2Filter.java]: Updated.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by dongjoon-hyun** [comment]
Thank you for updating, @panbingkun .

**Review by LuciferYang — APPROVED** [review]

**Comment by LuciferYang** [comment]
Merged into master for Spark 4.0. Thanks @panbingkun and @dongjoon-hyun ~

**Pull request closed without merging**